src/hidapi/libusb/hid.c

```diff
@@ -30,6 +30,18 @@
 #include "SDL_internal.h"
 #include "../../thread/SDL_systhread.h"
 
+#ifdef calloc
+#undef calloc
+#endif
+#define calloc	SDL_calloc
+#ifdef malloc
+#undef malloc
+#endif
+#define malloc	SDL_malloc
+#ifdef free
+#undef free
+#endif
+#define free	SDL_free
```

Comment:
Have you followed the code carefully to make sure that no SDL_malloc'ed memory is freed by libusb and no libusb-alloc'ed buffer is SDL_free'd?

Reply:
I read the code and the malloc/free are for allocating SDL-facing buffers/strings.
But since I don't have any joystick, I can't verify it can't verify this.
So it's probably safer to revert this commit.

Reply:
Feel free to open this as a separate PR, and I'll review.